src/transformers/models/mctc/configuration_mctc.py

```diff
@@ -0,0 +1,161 @@
+        intermediate_size (`int`, *optional*, defaults to 3072):
+            Dimension of the "intermediate" (i.e., feed-forward) layer in the Transformer encoder.
+        hidden_act (`str` or `function`, *optional*, defaults to `"gelu"`):
+            The non-linear activation function (function or string) in the encoder and pooler. If string, `"gelu"`,
+            `"relu"`, `"selu"` and `"gelu_new"` are supported.
+        hidden_dropout_prob (`float`, *optional*, defaults to 0.1):
+            The dropout probabilitiy for all fully connected layers in the embeddings, encoder, and pooler.
+        attention_probs_dropout_prob (`float`, *optional*, defaults to 0.1):
+            The dropout ratio for the attention probabilities.
+        max_position_embeddings (`int`, *optional*, defaults to 512):
+            The maximum sequence length that this model might ever be used with. Typically set this to something large
+            just in case (e.g., 512 or 1024 or 2048).
+        type_vocab_size (`int`, *optional*, defaults to 2):
+            The vocabulary size of the `token_type_ids` passed when calling [`~MCTCModel`] or [`~TFMCTCModel`].
+        initializer_range (`float`, *optional*, defaults to 0.02):
+            The standard deviation of the truncated_normal_initializer for initializing all weight matrices.
+        layer_norm_eps (`float`, *optional*, defaults to 1e-12):
+            The epsilon used by the layer normalization layers.
+        use_cache (`bool`, *optional*, defaults to `True`):
+            Whether or not the model should return the last key/values attentions (not used by all models). Only
+            relevant if `config.is_decoder=True`.
+        Example:
+
+    ```python
+    >>> from transformers import MCTCModel, MCTCConfig
+
+    >>> # Initializing a mCTC mctc-large style configuration
+    >>> configuration = MCTCConfig()
+
+    >>> # Initializing a model from the mctc-large style configuration
+    >>> model = MCTCModel(configuration)
+
+    >>> # Accessing the model configuration
+    >>> configuration = model.config
+    ```"""
+    model_type = "mctc"
+
+    def __init__(
+        self,
+        vocab_size=8065,
+        hidden_size=1536,
+        num_hidden_layers=36,
+        intermediate_size=6144,
+        num_attention_heads=4,
+        attention_head_dim=384,
+        position_embedding_type="relative_key",
+        max_position_embeddings=920,
+        layer_norm_eps=1e-5,
+        layerdrop=0.3,
+        hidden_act="relu",
+        initializer_range=0.02,
+        hidden_dropout_prob=0.3,
+        attention_probs_dropout_prob=0.3,
+        use_cache=True,
+        is_encoder_decoder=False,
+        pad_token_id=1,
+        bos_token_id=0,
+        eos_token_id=2,
+        conv_glu_dim=1,
+        conv_dropout=0.3,
+        num_conv_layers=1,
+        conv_kernel=[
+            7,
+        ],
+        conv_stride=[
```

Comment:
tuples are usually safer in signatures than lists :-)

Reply:
Not very important but I actually had used tuples in the beginning, but kept running into an error in testing because the tuple becomes a list when exporting to json. So importing the config from json imports it as a list and the [config  == export then import config from json test fails](https://app.circleci.com/pipelines/github/huggingface/transformers/39350/workflows/af7b0111-2a51-43c1-8b79-8911a457d890/jobs/441519?invite=true#step-111-5520)

I realized that `Wav2Vec2Config` dealt with this by changing the [tuple to list](https://github.com/huggingface/transformers/blob/main/src/transformers/models/wav2vec2/configuration_wav2vec2.py#L331) at `__init__` so fixed mine like so just now. Does make me wonder if `list` is more appropriate since thats what it ends up saving in the end.